### ISSUE TITLE
custom loss function

### ISSUE THREAD

**Issue by Sparkkkk** [tensorlayer/tensorlayer#721]
Do we have the flexibility to do our own loss function using python? 
I have written a custom loss function that combine cross entropy and a custom regularization term using y_ and output of some dense layers. But I haven't figured out how can I feed the value of a batch of sample into my custom regularization term.
